app/scripts/metamask-controller.js

```diff
@@ -5576,11 +5576,13 @@ export default class MetamaskController extends EventEmitter {
               },
             },
           ),
-        requestPermissionsForOrigin:
+        requestPermissionsForOrigin: (requestedPermissions) => {
           this.permissionController.requestPermissions.bind(
             this.permissionController,
             { origin },
-          ),
+            { [PermissionNames.permittedChains]: {}, ...requestedPermissions },
+          );
+        },
```

Comment:
we only want to add permittedChains if eth_accounts is requested. We actually also may want to do the other direction as well (add eth_accounts if permittedChains is requested)

Reply:
hmmm I suppose this is true. 

Currently the only permissions to request are `eth_accounts` or `permittedChains`. 

But if the request is made with no permissions we consider it malformed. So yeah we should check if the request contains an `eth_accounts` request and add this conditionally